amy/fiscal/tests/test_organization.py

```diff
@@ -48,24 +48,34 @@ def test_organization_delete(self):
                 Organization.objects.get(domain=org_domain)
 
     def test_organization_invalid_chars_in_domain(self):
-        r"""Ensure users can't put wrong characters in the organization's
-        domain field.
+        """Ensure organisation's domain is cleaned from URL scheme, if it was present.
+        Ensure other parts of the URL remain.
 
-        Invalid characters are any that match `[^\w\.-]+`, ie. domain is
-        allowed only to have alphabet-like chars, dot and dash.
-
-        The reason for only these chars lies in `workshops/urls.py`.  The regex
-        for the organization_details URL has `[\w\.-]+` matching...
+        The cleaning exists in OrganizationForm.clean_domain.
         """
-        data = {
-            "domain": "http://beta.com/",
-            "fullname": self.org_beta.fullname,
-            "country": self.org_beta.country,
-        }
-        url = reverse("organization_edit", args=[self.org_beta.domain])
-        rv = self.client.post(url, data=data)
-        # make sure we're not updating to good values
-        assert rv.status_code == 200
+        test_data = [
+            ("http://example.com/", "example.com/"),
+            ("https://example.com/", "example.com/"),
+            ("ftp://example.com/", "example.com/"),
+            ("http://example.com", "example.com"),
+            ("//example.com", "example.com"),
+            ("//example.com/", "example.com/"),
+            ("//example.com/?query", "example.com/?query"),
+            ("//example.com/path/", "example.com/path/"),
+            ("//example.com/path", "example.com/path"),
+            ("//example.com:80/path/?query", "example.com:80/path/?query"),
+            ("example.com/path;params?query#fragment", "example.com/path?query"),
+            (
+                "//user:password@example.com:80/path?query",
+                "user:password@example.com:80/path?query",
+            ),
+        ]
+        for domain, expected in test_data:
+            with self.subTest(domain=domain):
+                form = OrganizationForm({"domain": domain})
+                form.full_clean()
+                self.assertIn("domain", form.cleaned_data, form.errors)
```

Comment:
I think the form.errors is an accidental add.

Reply:
I sometimes debug the tests and forget to erase the last argument of `self.assert*` calls. It's for the message displayed when the tests fail; in this case I wanted to display form errors to diagnose form input data. All in all, this is pretty harmless, but I'll amend the commit to fix it.